src/components/Table/GridTable.test.tsx

```diff
@@ -2226,6 +2226,40 @@ describe("GridTable", () => {
     expect(sessionStorage.getItem(tableIdentifier)).toBe('["p2","p3"]');
   });
 
+  it("can lazily initialize table with collapsed rows", async () => {
+    // Given a table that can update its set of rows
+    function TestComponent() {
+      const [rows, setRows] = useState<GridDataRow<NestedRow>[]>(staticRows);
+      return (
+        <>
+          <button onClick={() => setRows(initRows)} data-testid="initRows" />
+          <GridTable columns={nestedColumns} rows={rows} fallbackMessage="Loading..." />
```

Comment:
Thought: We're not actually using `persistCollapse` here? The code changes seemed to be focus on "if `persistCollapse`", so I expected it in the test.

Reply:
The bug this fixed was happening when we lazily load rows regardless of `persistCollapse` being set. The `setRows` function was only checking for `initCollapsed` if it was **not** the first time loading "data" rows.
Previous if statement:
```
if (rows !== this.rows && this.rows.some((r) => r.kind !== "totals" && r.kind !== "header")) {
```

Reply:
Ah okay; I see we've got a separate test for the persistCollapse test; cool, looks good then. Thanks!